### PR TITLE
[ResourceGraph] Fetch subscriptions from ARM if AzureContext is not populated

### PR DESCRIPTION
**Pull request by ilidemi**
## Description

In Cloud Shell this.DefaultContext.Account is not populated with subscription ids by default. We want to detect that and fall back to ARM list subscriptions call, caching results within session.

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/ResourceManager/ResourceGraph/Commands.ResourceGraph/Az.ResourceGraph.psd1 [hunk 1]
```diff
@@ -51,7 +51,7 @@
         # ProcessorArchitecture = ''
 
         # Modules that must be imported into the global environment prior to importing this module
-        RequiredModules = @(@{ModuleName = 'Az.Profile'; ModuleVersion = '0.2.0'; })
+        RequiredModules = @(@{ModuleName = 'Az.Profile'; ModuleVersion = '0.2.2'; })
```
Comment (demyanenko) [inline src/ResourceManager/ResourceGraph/Commands.ResourceGraph/Az.ResourceGraph.psd1]: @cormacpayne The alias for Search-AzureRmGraph comes with 0.2.2, but if this change gets to the gallery before Cloud Shell updates Az.Profile, installing our package will break Cloud Shell again. Please advise which version would be better here.

Comment (cormacpayne) [reply src/ResourceManager/ResourceGraph/Commands.ResourceGraph/Az.ResourceGraph.psd1]: @demyanenko this change is fine

File: src/ResourceManager/ResourceGraph/Commands.ResourceGraph/Utilities/SubscriptionCache.cs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+                subscriptions.AddRange(page);
+
+                progressRecord.StatusDescription = $"Loading... {subscriptions.Count}";
+                parent.WriteProgress(progressRecord);
+            }
+            while (!string.IsNullOrEmpty(page.NextPageLink));
+            
+            // Making the total count visible
+            Thread.Sleep(100);
```
Comment (cormacpayne) [inline src/ResourceManager/ResourceGraph/Commands.ResourceGraph/Utilities/SubscriptionCache.cs]: @demyanenko we should remove this sleep if we don't _really_ need it 😀

**Comment by cormacpayne** [comment]
Desktop sign job: https://azuresdkci.westus2.cloudapp.azure.com/job/ps-sign/1397/
.NET Standard sign job: https://azuresdkci.westus2.cloudapp.azure.com/job/psnetsign/224/

File: src/ResourceManager/ResourceGraph/Commands.ResourceGraph/Utilities/SubscriptionCache.cs [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        /// <param name="parent">The parent.</param>
+        /// <param name="azureContext">The azure context.</param>
+        private static List<string> GetSubscriptionsWithProgressInternal(
+            Cmdlet parent, IAzureContext azureContext)
+        {
+            var subscriptionsClient =
+                AzureSession.Instance.ClientFactory.CreateArmClient<SubscriptionClient>(
+                    azureContext, AzureEnvironment.Endpoint.ResourceManager);
+
+            var progressRecord = new ProgressRecord(0, "Fetching subscriptions", "Loading...")
```
[the remark this answers is not in the document]
Comment (demyanenko) [reply src/ResourceManager/ResourceGraph/Commands.ResourceGraph/Utilities/SubscriptionCache.cs]: Removed the progress altogether. Getting subscriptions in Cloud Shell takes significantly less time than locally, so it's not really required.

File: src/ResourceManager/ResourceGraph/Commands.ResourceGraph/Az.ResourceGraph.psd1 (hunk 1 above)
Comment (demyanenko) [reply src/ResourceManager/ResourceGraph/Commands.ResourceGraph/Az.ResourceGraph.psd1]: @cormacpayne With 0.2.2 I'm not able to run it in current Cloud Shell without updating Az.Profile. Can we leave it at 0.2.0 and update after the shell has caught up? Or are there other ways to approach it?

**Comment by cormacpayne** [comment]
Desktop sign job: https://azuresdkci.westus2.cloudapp.azure.com/job/ps-sign/1404/
.NET Standard sign job: https://azuresdkci.westus2.cloudapp.azure.com/job/psnetsign/225/

**Comment by demyanenko** [comment]
@cormacpayne Works good in all environments, thanks for the review!

**Pull request merged**